### ISSUE TITLE
Delay between restart

### ISSUE THREAD

**Issue by yanickrochon** [remy/nodemon#1636]
## Feature Request

- `nodemon 1.19.2`:
- `node 12.10.0`:
- Operating system/terminal environment: Debian 10 
- Using Docker? No
- Command you ran: 
  `nodemon --delay 250ms --signal SIGHUP --exec "npx babel-node" src/app.js`

I need an option to let the app shutdown before restarting it; at the moment, the shutdown process is asynchronous, meaning that any terminating signal received is caught and handled by an asynchronous process, which then calls `process.exit`.

However, the app restarts too quickly, and this cause certain unwanted behaviors. Having a period between restart would solve this. For example, an option like 

```
nodemon --cooldown 300ms --signal SIGHUP --exec "npx babel-node" src/app.js
``

**Comment by remy** [comment]
Can you update to nodemon@2.0.1 and retest?

**Comment by Remzi1993** [comment]
Hi @yanickrochon 

Could you update nodemon by using the @latest tag, see:
```bash
npm install -g nodemon@latest
```
Or
```bash
npm install --save-dev nodemon@latest
```
https://bytearcher.com/articles/using-npm-update-and-npm-outdated-to-update-dependencies/
https://flaviocopes.com/update-npm-dependencies/

**Comment by yanickrochon** [comment]
Hi @Remzi1993 and @remy , thank you for notifying me about the update! I have tried to optimize my shutdown process also (before I updated) and I have read the latest changes. Since I am spawning different processes, they may have been part of the problem I had. In all case, the first time I launched the app with nodemon 2.0.1, the logs showed : 
``` 
[nodemon] app crashed - waiting for file changes before starting..."
```
but after killing it and restarting it, the error seem to have gone away. No idea what happened there. In all case, it seems to be properly shutting down now. Good job guys!
